### PR TITLE
Removed unnecessary complexity, spring cleanup

### PR DESCRIPTION
**Pull request by bilgili**

### CODE REVIEW

File: livre/core/render/GLContext.h [hunk 1]
```diff
@@ -43,9 +43,14 @@ class GLContext : public std::enable_shared_from_this< GLContext >
     LIVRECORE_API void shareContext( GLContextPtr dstContextPtr );
 
     /**
+     * @return a similar context.
```
Comment (jafyvilla) [inline livre/core/render/GLContext.h]: similar to what? A bit more of explanation would help.

File: livre/core/render/Renderer.h [hunk 2]
```diff
@@ -37,94 +37,66 @@ class Renderer
 
     /**
      * Renders the list of render bricks for a given frustum.
-     * @param glWidget Widget to render to.
-     * @param view The viewing information on the widget.
+     * @param frustum Viewfrustum
```
Comment (jafyvilla) [inline livre/core/render/Renderer.h]: Same here, this does not say much at a first glance

File: livre/eq/Channel.cpp [hunk 3]
```diff
@@ -307,6 +238,33 @@ struct Channel::Impl
...
+
+        DashRenderNode renderNode( dashNode );
+        if( renderNode.getLODNode().getRefLevel() != 0 )
+            renderNode.setTextureObject( CacheObjectPtr( ));
+    }
+
+    void freeTextures()
+    {
+        for( const ConstCacheObjectPtr& cacheObject:
+                       _frameInfo.renderNodes )
```
Comment (jafyvilla) [inline livre/eq/Channel.cpp]: Fits in one line

File: livre/core/render/GLContext.h (hunk 1 above)
Comment (bilgili) [inline livre/core/render/GLContext.h]: I think i should be named clone(), makes much more sense

File: livre/core/render/Renderer.h (hunk 2 above)
Comment (bilgili) [inline livre/core/render/Renderer.h]: The function explanation ?

Comment (jafyvilla) [inline livre/core/render/Renderer.h]: The param sections, specially "frustum"

File: livre/core/render/Renderer.cpp [hunk 4]
```diff
@@ -47,69 +46,43 @@ struct DistanceOperator
...
 
-void Renderer::_order( RenderBricks &bricks, const Frustum &frustum ) const
-{
     DistanceOperator distanceOp( frustum );
-    std::sort( bricks.begin(), bricks.end(), distanceOp );
-}
+    std::sort( rbs.begin(), rbs.end(), distanceOp );
 
+     for( auto& rb: rbs )
+        bricksDst.push_back( *rb );
```
Comment (eile) [inline livre/core/render/Renderer.cpp]: std::copy (twice in function)?

Comment (eile) [inline livre/core/render/Renderer.cpp]: Actually:

```RenderBricks _order( const RenderBricks& bricks, const Frustum& frustum ) const```

Sort on RenderBricks ret; directly -- why copy into vector< * >?

Comment (eile) [inline livre/core/render/Renderer.cpp]: Or alternatively: ```void _order( RenderBricks& bricks, const Frustum& frustum ) const```

Comment (bilgili) [inline livre/core/render/Renderer.cpp]: Done

File: livre/eq/Channel.cpp (hunk 3 above)
Comment (bilgili) [inline livre/eq/Channel.cpp]: Done

**Comment by jafyvilla** [comment]
+1

**Comment by bilgili** [comment]
please retest this

**Comment by bilgili** [comment]
retest this please

**Pull request merged**